include/aws/sdkutils/endpoints/ruleset.h

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#ifndef AWS_SDKUTILS_ENDPOINTS_RULESET_H
+#define AWS_SDKUTILS_ENDPOINTS_RULESET_H
+
+#include <aws/common/byte_buf.h>
+#include <aws/sdkutils/sdkutils.h>
+
+struct aws_endpoints_ruleset;
+struct aws_endpoints_parameter;
+struct aws_hash_table;
+
+enum aws_endpoints_parameter_value_type { AWS_ENDPOINTS_PARAMETER_STRING, AWS_ENDPOINTS_PARAMETER_BOOLEAN };
+
+AWS_EXTERN_C_BEGIN
+
+/*
+******************************
+* Parameter
+******************************
+*/
+
+/*
+ * Value type of parameter.
+ */
+AWS_SDKUTILS_API enum aws_endpoints_parameter_value_type aws_endpoints_parameter_get_value_type(
+    const struct aws_endpoints_parameter *parameter);
```

Comment:
Rulesets are versioned and do not make forwards compat guarantees.
There is a version check in parser (thats currently disabled, because test files are not consistent with versioning). And parsing should fail if it encounters anything thats it does not know about. 
My vote here is to fail parsing and not try to recover, but happy to discuss further.